### ISSUE TITLE
bug: TypeScript imports are scrambled together

### ISSUE THREAD

**Issue by buehler** [prettier/prettier-vscode#608]
Hey there, I'm using VSCode `v1.28.2` and prettier-vscode `v1.6.1`.

Since a period of time, I'm encountering the following bug when prettier is reformating imports:

First, the imports are like this:
<img width="300" alt="2018-10-24t08-35-16_90c0fdb2" src="https://user-images.githubusercontent.com/1067213/47411212-4e001900-d768-11e8-8802-2f90c0d2e4fb.png">

Now, my linting rules imply, that you order the imports in the same group. And prettier does the following job :-)

<img width="299" alt="2018-10-24t08-39-03_bd87279d" src="https://user-images.githubusercontent.com/1067213/47411239-5eb08f00-d768-11e8-9d77-e1a92d2108f8.png">

Does somebody else have this exact problem?
Cheers

**Comment by RobinMalfait** [comment]
This is beautiful, consider it **Art**.

For reals: I think there is a race condition going on somewhere but I don't know what it could be. 🤔

**Comment by CiGit** [comment]
Can you share your settings ?
I suspect tslint with sort imports and organize import on save

**Comment by buehler** [comment]
Hmm that could be... which settings do you need? the ones for this extension or tslint json?

**Comment by buehler** [comment]
```json
{
    "prettier.printWidth": 125,
    "prettier.singleQuote": true,
    "prettier.stylelintIntegration": true,
    "prettier.trailingComma": "es5",
    "prettier.tslintIntegration": true,
    "prettier.arrowParens": "always"
}
```

**Comment by CiGit** [comment]
Things to test:
- Call format Document instead of FormatOnSave
- Disable tslintIntegration and call tslint --fix
- I can't find it, but It seems there is something like _Organize imports on save_

**Comment by josephktcheung** [comment]
This bug should be fixed once we update `prettier-tslint` to `0.4.2` https://github.com/azz/prettier-tslint/issues/26
